chain/cardano/client/client.go

```diff
@@ -0,0 +1,407 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"maps"
+	"net/http"
+	"slices"
+	"time"
+
+	xc "github.com/cordialsys/crosschain"
+	xcbuilder "github.com/cordialsys/crosschain/builder"
+	"github.com/cordialsys/crosschain/chain/cardano/client/types"
+	"github.com/cordialsys/crosschain/chain/cardano/tx_input"
+	xclient "github.com/cordialsys/crosschain/client"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	POST = "POST"
+	GET  = "GET"
+	// Cardano uses lovelace as the smallest unit of ada
+	// 1 lovelace = 0.000001 ada
+	Lovelace    = "lovelace"
+	Ada         = "ADA"
+	API_VERSION = "/api/v0"
+)
+
+// Client for Template
+type Client struct {
+	Asset               xc.ITask
```

Comment:
To guard against using `FeeLimit` parameter mistake in client, perhaps change construction:

```go
type Client struct {
	Cfg               *xc.ChainClientConfig
        NativeAsset xc.NativeAsset
       // (other base options copied manually)
// ...
}
```